openapi/task_execution_service.openapi.yaml

```diff
@@ -100,6 +100,55 @@ paths:
           If unspecified, no task name filtering is done.
         schema:
           type: string
+      - name: state
+        description: |-
+          OPTIONAL. Filter tasks by state. If unspecified,
+          no task state filtering is done.
+        in: query
+        required: false
+        schema:
+          $ref: '#/components/schemas/tesState'
+      - name: tag_key
+        description: |-
+          OPTIONAL. Provide key tag to filter. The field tag_key is an array
+          of key values, and will be zipped with an optional tag_value array.
+          So the query:
+          ```
+            ?tag_key=foo1&tag_value=bar1&tag_key=foo2&tag_value=bar2
```

Comment:
What if the ordering of `tag_key` and `tag_value` params is not interleaved like in this example? I suppose the `tag_key` and `tag_value` arrays could be independently constructed and then zipped. But what if they aren't of equal length?

Also, is it possible to specify deliberately/explicitly an empty value, and if so, how? Would this be expected to work: `?tag_key=foo1&tag_value=&tag_key=food2&tag_value=bar2`? And if so, how about the same for keys, would an empty key be allowed?

Finally, there aren't any guidelines on escaping characters. What if a key or value name includes characters such as `?` or `&`?

Reply:
The `tag_key` and `tag_value` arrays would be constructed first, and 'zipped' together. The shorter array would be filled in with blanks and behave in the way described on line 126.

You are correct, treating empty strings as 'wildcards' means that you can't search specifically for empty strings. I'm not really sure of an alternative, but I'm not sure if this is a blocker, as you could simple search for the existence of the tag and filter values manually on the client side.

I think character escaping would work via traditional HTML escaping. IE `?` would be `&#63;` and `&` would be `&amp;`

Reply:
@kellrott  so for this example below what is the expected behavior?

`{"foo": "", "baz": ""} `

Reply:
For the input `{"foo": "", "baz": ""}` we expect that both keys `foo` and `baz` exist. The values will be ignored and all instances of tasks with both `foo` and `baz` keys defined will be returned. 

This could be coded as `?tag_key=foo&tag_key=baz` with no `tag_value` set

Reply:
Thanks. This LGTM.

Reply:
Fine with me too :)